group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -8117,7 +8112,7 @@ public List<DeleteShareGroupStateRequestData.DeleteStateData> sharePartitionsEli
                     // If the topic for which delete share group offsets request is sent is already present in the deletingTopics set,
                     // we will include that topic in the delete share group state request.
                     List<DeleteShareGroupStateRequestData.PartitionData> partitions = new ArrayList<>();
-                    topicImage.partitions().keySet().forEach(partition ->
+                    IntStream.range(0, topicMetadata.partitionCount()).boxed().collect(Collectors.toSet()).forEach(partition ->
```

Comment:
This looks like a mechanical refactoring that went a bit wrong. We can do:

```
                    List<DeleteShareGroupStateRequestData.PartitionData> partitions = IntStream.range(0, topicMetadata.partitionCount())
                            .mapToObj(partition -> new DeleteShareGroupStateRequestData
                                    .PartitionData()
                                    .setPartition(partition))
                            .toList();
```
instead.

Reply:
ah good catch, yea that was some copy pasta from removing the built-in `partitions()` method from the CoordinatorMetadata interface